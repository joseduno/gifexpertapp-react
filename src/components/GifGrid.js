import React from 'react'
import GifGridItem from './GifGridItem'
import {useFetchGifs} from '../hooks/useFetchGifts'
import PropTypes from 'prop-types'


const GifGrid = ({category}) => {
    
    const {data:images, loading} = useFetchGifs(category)

    return (
        <div className='card-grid'>
            {/* con {...img} pasamos cada propiedad y valor de la imagen como 
            argumentos individuales al componente GifGridItem */}
            {loading && <p className='animate__animated animate__heartBeat'>Cargando...</p>}
            {images.map(img => <GifGridItem key={img.id} {...img} />)}
        </div>
    )
}


GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}


export default GifGrid
