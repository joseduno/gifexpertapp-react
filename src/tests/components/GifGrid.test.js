import '@testing-library/jest-dom'
import { shallow } from "enzyme"
import GifGrid from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifts'


jest.mock('../../hooks/useFetchGifts')


describe('Pruebas sobre el componente GifGrid', () => {
    const category = 'Godzilla'
    const gifs = [{
        id: 'ABC',
        url: 'https://localhost/cualquier/cosa.jpg',  // no importa si no existe
        title: 'Cualquier cosa'
    },{
        id: '123',
        url: 'https://localhost/cualquier/cosa2.jpg',  // no importa si no existe
        title: 'Cualquier cosa 2'
    }]

    test('debe renderizar correctamente el componente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
        // Se debe renderizar el componente luego de simular el uso de los custom hooks
        // si la llamada al shallow ocurre fuera del test, NO FUNCIONA
        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot()
    })

    test('debe mostrar items cuando se termine la peticion de gifs', () => {
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })
        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper.find('p').exists()).toBe(false)
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
    })
    
})
