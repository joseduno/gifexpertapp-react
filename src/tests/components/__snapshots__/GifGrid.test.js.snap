// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente GifGrid debe renderizar correctamente el componente 1`] = `
<div
  className="card-grid"
>
  <p
    className="animate__animated animate__heartBeat"
  >
    Cargando...
  </p>
</div>
`;
